### PR TITLE
chore: set minimum specification for codespace machines

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Tell GitHub to use at least 4 core machines with at least 8GB of RAM and 32 GB of storage when using Codespaces

## Context

Running `yarn test` on the default 2 core machine doesn't work, as the test fail (probably due to heap exhaustion). I tested the 4 core machine on Codespaces, which works fine.

Here's the GitHub docs:

https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/setting-a-minimum-specification-for-codespace-machines

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
what are the specs of the default machine?

we set 500MB for the jest worker by default
https://github.com/renovatebot/renovate/blob/c11b8c4561c7ec9767a406cb93714e75d93a02f6/jest.config.ts#L89

we can set another limit for codespaces as we do for github actions

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,4 +1,9 @@
 {
+  "hostRequirements": {
+    "cpus": 4,
+    "memory": "8gb",
+    "storage": "32gb"
```
Comment (rarkins) [inline .devcontainer/devcontainer.json:5]: Do we actually use that much? Or it's just so that we have plenty in future if running `yarn start ...` ?

Comment (HonkingGoose) [reply .devcontainer/devcontainer.json:5]: I think 32 GB or storage is the default for the 4 core tier?

**Comment by rarkins** [comment]
> what are the specs of the default machine?
> 
> we set 500MB for the jest worker by default
> 
> https://github.com/renovatebot/renovate/blob/c11b8c4561c7ec9767a406cb93714e75d93a02f6/jest.config.ts#L89
> 
> we can set another limit for codespaces as we do for github actions

Do you mean this?

```
Host stats:
  Cpus:      2
  Memory:    3.84 GB
  HeapLimit: 1.97 GB
```

For 4 CPUs the RAM was doubled but heap unchanged.

**Pull request merged**